src/components/AttachmentModal.js

```diff
@@ -230,9 +230,14 @@ class AttachmentModal extends PureComponent {
     }
 
     render() {
-        const sourceURL = this.props.isAuthTokenRequired
-            ? addEncryptedAuthTokenToURL(this.state.sourceURL)
-            : this.state.sourceURL;
+        let source = this.state.source;
+
+        // if source is a URL, add auth token to get access
+        if (_.isString(this.state.source)) {
+            source = this.props.isAuthTokenRequired
+                ? addEncryptedAuthTokenToURL(this.state.source)
+                : this.state.source;
+        }
```

Comment:
How about we simplify this a bit like this?

```suggestion
        const source = _.isString(this.state.source) && this.props.isAuthTokenRequired
            ? addEncryptedAuthTokenToURL(this.state.source)
            : this.state.source;
```